extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {user} from '../../../src/log';
+
+
+/** @const */
+// const EXPERIMENT = 'amp-story-auto-ad';
+
+/** @const */
+const TAG = 'amp-story-auto-ads';
+
+export class AmpStoryAutoAds extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+    this.ampStory = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    const ampStory = this.element.parentElement;
+    user().assert(ampStory.tagName === 'AMP-STORY',
+        `<${TAG}> should be child of <amp-story>`);
+    this.ampStory = ampStory;
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    return true;
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.ampStory.appendChild(this.makeMockPage());
```

Comment:
LGTM for now.  I don't think this is going to be sufficient long-term, since the amp-story won't know its children have changed.  We'll need to devise a way to tell the story that it has a new page.  You could either dispatch an event, or expose a function from AmpStory and call it here, like:

    ampStory.getImpl().then(storyImpl => {
      storyImpl.addNewPage(this.makeMockPage());
    });

Reply:
In long term, maybe worth to extract a PageManager class to decouple the dependency.

Reply:
👍